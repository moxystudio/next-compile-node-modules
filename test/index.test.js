'use strict';

const compileNodeModulesPlugin = require('../index');

const createWebpackConfig = () => ({
    module: {
        rules: [
            {
                test: /\.(tsx|ts|js|mjs|jsx)$/,
                include: [
                    '/path/to/project',
                    /next[\\/]dist[\\/]next-server[\\/]lib/,
                    /next[\\/]dist[\\/]client/,
                    /next[\\/]dist[\\/]pages/,
                    /[\\/](strip-ansi|ansi-regex)[\\/]/,
                ],
                exclude: () => {},
                use: {
                    loader: 'next-babel-loader',
                    options: {
                        isServer: true,
                        hasModern: false,
                        distDir: '/path/to/project/.next',
                        cwd: '/path/to/project',
                        cache: true,
                    },
                },
            },
        ],
    },
});

it('should duplicate the default JS rule', () => {
    const config = compileNodeModulesPlugin()().webpack(createWebpackConfig());

    expect(config.module.rules).toMatchSnapshot();
});

it('should throw if the JS rule was not found', () => {
    const noGoodRuleConfig = createWebpackConfig();

    noGoodRuleConfig.module.rules[0].use.loader = 'bad-loader';

    expect(() => compileNodeModulesPlugin()().webpack(noGoodRuleConfig)).toThrow(/JS rule/);
});

it('should allow to override the rule\'s test, include and exclude properties', () => {
    const options = {
        test: 'my-test',
        include: 'my-include',
        exclude: 'my-exclude',
    };

    const config = compileNodeModulesPlugin(options)().webpack(createWebpackConfig());

    expect(config.module.rules).toMatchSnapshot();
});

it('should call nextConfig webpack if defined', () => {
    const nextConfig = {
        webpack: jest.fn(() => 'foo'),
    };

    const config = compileNodeModulesPlugin()(nextConfig).webpack(createWebpackConfig());

    expect(nextConfig.webpack).toHaveBeenCalledTimes(1);
    expect(config).toBe('foo');
});
