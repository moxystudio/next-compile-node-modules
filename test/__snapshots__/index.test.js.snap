// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should allow to override the rule's test, include and exclude properties 1`] = `
Array [
  Object {
    "exclude": [Function],
    "include": Array [
      "/path/to/project",
      /next\\[\\\\\\\\\\\\/\\]dist\\[\\\\\\\\\\\\/\\]next-server\\[\\\\\\\\\\\\/\\]lib/,
      /next\\[\\\\\\\\\\\\/\\]dist\\[\\\\\\\\\\\\/\\]client/,
      /next\\[\\\\\\\\\\\\/\\]dist\\[\\\\\\\\\\\\/\\]pages/,
      /\\[\\\\\\\\\\\\/\\]\\(strip-ansi\\|ansi-regex\\)\\[\\\\\\\\\\\\/\\]/,
    ],
    "test": /\\\\\\.\\(tsx\\|ts\\|js\\|mjs\\|jsx\\)\\$/,
    "use": Object {
      "loader": "next-babel-loader",
      "options": Object {
        "cache": true,
        "cwd": "/path/to/project",
        "distDir": "/path/to/project/.next",
        "hasModern": false,
        "isServer": true,
      },
    },
  },
  Object {
    "exclude": "my-exclude",
    "include": "my-include",
    "test": "my-test",
    "use": Object {
      "loader": "next-babel-loader",
      "options": Object {
        "cache": true,
        "caller": Object {
          "isNodeModule": true,
        },
        "cwd": "/path/to/project",
        "distDir": "/path/to/project/.next/compile-node-modules",
        "hasModern": false,
        "isServer": true,
      },
    },
  },
]
`;

exports[`should duplicate the default JS rule 1`] = `
Array [
  Object {
    "exclude": [Function],
    "include": Array [
      "/path/to/project",
      /next\\[\\\\\\\\\\\\/\\]dist\\[\\\\\\\\\\\\/\\]next-server\\[\\\\\\\\\\\\/\\]lib/,
      /next\\[\\\\\\\\\\\\/\\]dist\\[\\\\\\\\\\\\/\\]client/,
      /next\\[\\\\\\\\\\\\/\\]dist\\[\\\\\\\\\\\\/\\]pages/,
      /\\[\\\\\\\\\\\\/\\]\\(strip-ansi\\|ansi-regex\\)\\[\\\\\\\\\\\\/\\]/,
    ],
    "test": /\\\\\\.\\(tsx\\|ts\\|js\\|mjs\\|jsx\\)\\$/,
    "use": Object {
      "loader": "next-babel-loader",
      "options": Object {
        "cache": true,
        "cwd": "/path/to/project",
        "distDir": "/path/to/project/.next",
        "hasModern": false,
        "isServer": true,
      },
    },
  },
  Object {
    "exclude": undefined,
    "include": /\\[\\\\\\\\\\\\/\\]node_modules\\[\\\\\\\\\\\\/\\]/,
    "test": /\\\\\\.js\\$/,
    "use": Object {
      "loader": "next-babel-loader",
      "options": Object {
        "cache": true,
        "caller": Object {
          "isNodeModule": true,
        },
        "cwd": "/path/to/project",
        "distDir": "/path/to/project/.next/compile-node-modules",
        "hasModern": false,
        "isServer": true,
      },
    },
  },
]
`;
